src/Components/WebAssembly/Build/src/Tasks/GzipCompressBlazorApplicationFiles.cs

```diff
@@ -33,7 +33,7 @@ public override bool Execute()
                 var targetCompressionPath = file.Target;
 
                 if (!File.Exists(inputSource) ||
-                    (File.Exists(targetCompressionPath) && File.GetLastWriteTime(inputSource) > File.GetLastWriteTime(targetCompressionPath)))
+                    (File.Exists(targetCompressionPath) && File.GetLastWriteTime(inputSource) < File.GetLastWriteTime(targetCompressionPath)))
                 {
                     // Incrementalism. If input source doesn't exist or it exists and is not newer than the expected output, do nothing.
                     if (!File.Exists(inputSource))
```

Comment:
Might be easier to rewrite it like this and remove the outer condition. This avoids two file lookups per file that hasn't changed.

```suggestion
if (!File.Exists(inputSource))
{
    Log.LogMessage($"Skipping '{inputPath}' because '{inputSource}' does not exist.");
    return;
}
else if ((File.Exists(targetCompressionPath) && File.GetLastWriteTimeUtc(inputSource) < File.GetLastWriteTimeUtc(targetCompressionPath)))
{
    Log.LogMessage($"Skipping '{inputPath}' because '{targetCompressionPath}' is newer than '{inputSource}'.");
    return;
}
```

Note that the LogMessage is incorrect now (even if you don't choose to use this code).